lib/ash/reactor/steps/destroy_step.ex

```diff
@@ -58,18 +60,12 @@ defmodule Ash.Reactor.DestroyStep do
       [return_notifications?: false]
       |> maybe_set_kw(:authorize?, options[:authorize?])
 
-    attributes =
-      options[:resource]
-      |> Ash.Resource.Info.attributes()
-      |> Map.new(&{&1.name, record[&1]})
-      |> then(&Map.merge(arguments[:input], &1))
-
     options[:resource]
-    |> Changeset.for_create(options[:undo_action], attributes, changeset_options)
+    |> Changeset.for_create(options[:undo_action], %{record: record}, changeset_options)
```

Comment:
we should also set the changeset metadata on this record right?

Reply:
Should it? Does a changeset for `destroy` contain anything useful?  The only reason we do it on the `update` step is because it's the only way to pass the changeset to the `undo` function if we need it later.

Reply:
A destroy action can be "soft" in which case it is an update masquerading as a destroy.